Provenance Graph

utku-ozturk opened the pull request.

utku-ozturk committed:
bump react-workflow-viz 0.2.0-beta.1

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
add links to chart and workflow visualization scss

utku-ozturk committed:
visualization and workflow scss

utku-ozturk committed:
add missing `workflow_run_inputs` and `workflow_run_outputs` rev fields

utku-ozturk committed:
add `MetaWorkflowRunView`view to register

utku-ozturk committed:
add metaworkflowrun embeds

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
get file_format from encoded_core

utku-ozturk committed:
get app_name and steps from encoded_core/workflow

utku-ozturk committed:
get quality_metric from encoded_core/file

utku-ozturk committed:
add MetaWorkflowRun item page having graph tab

utku-ozturk committed:
default node renderer for wf viz

utku-ozturk committed:
add provenance graph tab to file overview page

utku-ozturk committed:
add workflow detail pane

utku-ozturk committed:
update `Workflow` type by adding trace_meta_workflows end point

utku-ozturk committed:
add controller to fetch workflow steps

utku-ozturk committed:
add workflow graph toolbar

utku-ozturk committed:
add graph options state controller

utku-ozturk committed:
add provenance graph tab

utku-ozturk committed:
type `WorkflowRun` updates

utku-ozturk committed:
move visualization registration from include_snovault to include_encoded

utku-ozturk committed:
Legend - currently not functional

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
temporarily comment out `meta_workflow.workflows.input.value` embed

utku-ozturk committed:
add creating an account doc link into login popup

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
add qc embeds

utku-ozturk committed:
check quality_metrics array in addition to quality_metric

utku-ozturk committed:
fix detail modal's title overflow

utku-ozturk committed:
rebuild tooltip

utku-ozturk committed:
remove shadow

utku-ozturk committed:
add data_generation_summary and accession embeds

utku-ozturk committed:
add legend and basic I/O option

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
high contrast

utku-ozturk committed:
hide empty columns

utku-ozturk committed:
decrease settings label font size

utku-ozturk committed:
rename components

utku-ozturk committed:
more refactoring

utku-ozturk committed:
fix typo

utku-ozturk committed:
more refactoring

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
bug fix for fillMissingChildBuckets

utku-ozturk committed:
disable file downloads on MetaWorkflowRun item page using new `canDownloadFile` prop

utku-ozturk committed:
reduce MWFR embeds

utku-ozturk committed:
temporarily opt-out the embeds

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
fix search type and update default date interval

utku-ozturk committed:
show usage tab filter/settings always

utku-ozturk committed:
change type as SubmittedFile

utku-ozturk committed:
bind this to getXAxisGenerator

utku-ozturk committed:
refactor y axis scale + add y axis scaling into submission statistics

utku-ozturk committed:
add missing dates

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
bug fix and rewrite date range presets

utku-ozturk committed:
adjust submission filters width

utku-ozturk committed:
get empty buckets for missing dates

utku-ozturk committed:
add date interval related fields to response + update date range presets

utku-ozturk committed:
add missing dates for submissions

utku-ozturk committed:
fix weekly missing dates insert

utku-ozturk committed:
date range updates

utku-ozturk committed:
add missing children in submissions tab chart

utku-ozturk committed:
disable modal for non-admins

utku-ozturk committed:
remove no longer used externalTermMap

utku-ozturk committed:
add date histogram interval option into filter

utku-ozturk committed:
basic document formatting

utku-ozturk committed:
add from date for option `all`

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
update term text

utku-ozturk committed:
change button variant to outline-secondary

utku-ozturk committed:
override sticky first column's style

utku-ozturk committed:
horizontally center y-axis scale controls

utku-ozturk committed:
adjust col widths of settings

utku-ozturk committed:
high/mid dpi screen adjustments

utku-ozturk committed:
update toggle style

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
fix typo

utku-ozturk committed:
change type=SubmittedFile with File for released  in submissions statistics

utku-ozturk committed:
bump react-workflow-viz

utku-ozturk committed:
change default graph scale

utku-ozturk committed:
fix typo

utku-ozturk committed:
remove unused code blocks

utku-ozturk committed:
exclude HMS DAC for meta data/data submitted

utku-ozturk committed:
add GCC-only and TTD-only options

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
feat: enhance workflow visualization with grouped file handling and UI improvements

utku-ozturk committed:
feat: add MetaWorkflowView component and refactor identifiersToIntegers function

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
fix: pass request parameter to CoreWorkflowRun.steps in WorkflowRun class

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
feat: add React Workflow Viz build and watch tasks to Gulp

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

dbmi-svc-checkmarx commented:
![Logo](https://cdn.ast.checkmarx.net/integrations/logo/Checkmarx.png)
**Checkmarx One – Scan Summary & Details** – [4f24efac\-fc23\-4102\-bb4a\-0cfa9657fba9](https://us.ast.checkmarx.net/projects/e9cf671f-a7fd-493f-900d-663a40a13bee/scans?id=4f24efac-fc23-4102-bb4a-0cfa9657fba9&branch=utk_provenance_graph)
### Great job! No new security vulnerabilities introduced in this pull request
<!--footer section-->

---
Use @Checkmarx to interact with Checkmarx PR Assistant. 
*Examples:*
`@Checkmarx how are you able to help me?`
`@Checkmarx rescan this PR`
<!--footer section-->

utku-ozturk committed:
feat: enhance file size and format display in workflow components

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
Enhance Workflow Visualization: Introduce Compact Mode, Improved Node Titles, and Styling Adjustments

- Implemented a compact mode for the workflow graph to improve readability, allowing users to toggle between basic and detailed views.
- Added functions to extract and display file formats, and to determine auxiliary nodes for better categorization.
- Enhanced node title rendering with human-readable formats and compact titles for steps.
- Introduced badges for merged nodes and quality metrics, improving visual feedback on node status.
- Updated styling for nodes, edges, and legends to enhance the overall user interface and experience.
- Adjusted layout and spacing for better responsiveness and usability across different screen sizes.

utku-ozturk committed:
Enhance Workflow Node Functionality and Styling: Add Primary Data Node Check, Merge Connectivity Logic, and Improve Color Contrast for Node Categories

utku-ozturk committed:
Enhance Workflow Visualization: Improve Node Grouping Logic, Update Step Group Labels, and Refine Color Palette for Accessibility

utku-ozturk committed:
Enhance Workflow Graph: Add showDetailsInPopup state to commonGraphProps and update dimNonPathOnSelect logic

utku-ozturk committed:
Enhance Workflow Node Title Display: Improve step title logic to include node title fallback

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
Merge branch 'main' into utk_provenance_graph

utku-ozturk committed:
feat: enhance workflow graph rendering by implementing node column reassignment and merging duplicate file nodes
